scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -2911,7 +2966,7 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
    */
   def blockStatSeq(): List[Stat] = {
     val stats = new ListBuffer[Stat]
-    while (!token.is[StatSeqEnd] && !token.is[CaseDefEnd]) {
+    while (!token.is[StatSeqEnd] && !token.is[CaseDefEnd] && !ahead(token.is[`case`])) {
```

Comment:
One of tests didn't pass, and analysis showed that issue can be resolved by adding that checking.